avocado/core/test.py

```diff
@@ -877,37 +879,50 @@ def _report(self):
             self.log.info("%s %s", self.status,
                           self.name)
 
-    def fail(self, message=None):  # pylint: disable=W0221
+    def _deprecate_params_message(func):  # pylint: disable=E0213
+        """ This decorator helps to deprecate parameter 'message' and
+        replace it with 'msg'.
+        """
+
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+            message = kwargs.get('message', None)
+            if message is not None:
+                kwargs['msg'] = message
+                warnings.warn("The parameter 'message' has been deprecated. Please use instead 'msg'", DeprecationWarning)
```

Comment:
One corner case not handled here seems to be if the user gives both `msg` and `message` parameters, with `msg` now being a possibility in either in positional or keyword form.  I think we can simplify the decorated methods, while still covering this corner case.  One positive aspect is that their signature will remain the same for the future, when the decorators are removed.

Suggestion, on top of your work:

```diff
diff --git a/avocado/core/test.py b/avocado/core/test.py
index 23a896448..20e8699ef 100644
--- a/avocado/core/test.py
+++ b/avocado/core/test.py
@@ -886,15 +886,32 @@ class Test(unittest.TestCase, TestData):
 
         @functools.wraps(func)
         def wrapper(*args, **kwargs):
-            message = kwargs.get('message', None)
-            if message is not None:
-                kwargs['msg'] = message
-                warnings.warn("The parameter 'message' has been deprecated. Please use instead 'msg'", DeprecationWarning)
-            return func(*args, **kwargs)
+            msg = None
+            message = None
+
+            if 'msg' in kwargs:
+                msg = kwargs.get('msg')
+            elif args:
+                msg = args[0]
+
+            if 'message' in kwargs:
+                message = kwargs.get('message')
+                if msg:
+                    warnings.warn("Please favor parameter 'msg' and do not use "
+                                  "it with the parameter 'message'",
+                                  DeprecationWarning)
+
+                else:
+                    warnings.warn("The parameter 'message' has been deprecated."
+                                  " Please use 'msg' instead", DeprecationWarning)
+
+            actual_message = msg or message
+            return func(actual_message)
         return wrapper
 
+    @staticmethod
     @_deprecate_params_message
-    def fail(self, msg=None, **kwargs):
+    def fail(msg=None):
         """
         Fails the currently running test.
 
@@ -908,7 +925,7 @@ class Test(unittest.TestCase, TestData):
 
     @staticmethod
     @_deprecate_params_message
-    def error(msg=None, **kwargs):
+    def error(msg=None):
         """
         Errors the currently running test.
 
@@ -922,7 +939,7 @@ class Test(unittest.TestCase, TestData):
 
     @staticmethod
     @_deprecate_params_message
-    def cancel(msg=None, **kwargs):
+    def cancel(msg=None):
         """
         Cancels the test.
 

```

I've tested it with this:

```python
from avocado import Test


class FailTest(Test):

    def test_empty(self):
        self.fail()

    def test_positional(self):
        self.fail('msg positional')

    def test_msg_keyword(self):
        self.fail(msg='msg keyword')

    def test_message_keyword(self):
        self.fail(message='msg keyword')

    def test_msg_message(self):
        self.fail('msg positional', message='message keyword')

```

Reply:
Thanks @clebergnu , I like this one more!